### PR TITLE
Traces: APM table improvements

### PR DESCRIPTION
**Pull request by joey-grafana**
**What this PR does / why we need it**:

This PR adds improvements to the queries used in the APM table.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: public/app/plugins/datasource/tempo/graphTransform.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export const errorRateMetric = {
 };
 export const durationMetric = {
   expr: 'histogram_quantile(.9, sum(rate(traces_spanmetrics_duration_seconds_bucket{}[$__range])) by (le))',
-  params: ['span_status="STATUS_CODE_ERROR"'],
+  params: [],
```
Comment (joe-elliott) [inline public/app/plugins/datasource/tempo/graphTransform.ts:149]: i assume the filter params are added elsewhere?

File: public/app/plugins/datasource/tempo/datasource.ts [hunk 2]
```diff
@@ -766,6 +766,8 @@ export function buildExpr(
 }
 
 export function buildLinkExpr(expr: string) {
+  // don't want top 5 or by span name in links
+  expr = expr.replace('topk(5, ', '').replace(' by (span_name))', '');
```
Comment (joe-elliott) [inline public/app/plugins/datasource/tempo/datasource.ts:770]: unlikely but what happens if someone applies a filter with one of these substrings. i.e. at the top they have selected something silly like `nameapace="topk(5, "`

**Review by joe-elliott — COMMENTED** [review]
Couple of thoughts, but on the surface this appears to do what was requested:

- remove sum by span name and topk from the linked prometheus query
- remove span_status="STATUS_CODE_ERROR" from both the latencies in the table and from the linked prometheus query

File: public/app/plugins/datasource/tempo/graphTransform.ts (hunk 1 above)
Comment (joey-grafana) [reply public/app/plugins/datasource/tempo/graphTransform.ts:149]: Yeah, in the buildExpr function. Helps me to merge params, filters, and __data.fields for the query :)

File: public/app/plugins/datasource/tempo/datasource.ts (hunk 2 above)
Comment (joey-grafana) [reply public/app/plugins/datasource/tempo/datasource.ts:770]: We wouldn't reach this part of the code as the query wouldn't return any frames. Ideally we should not let users input anything into the filter inputs other that what is already in the dropdown?

**Comment by joey-grafana** [comment]
> Couple of thoughts, but on the surface this appears to do what was requested:
> 
> * remove sum by span name and topk from the linked prometheus query
> * remove span_status="STATUS_CODE_ERROR" from both the latencies in the table and from the linked prometheus query

Sorry, are both of these requests solely in the APM table?
- I thought I had removed the topk and by span name from the [queries already](https://github.com/grafana/grafana/pull/50969/files#diff-9c80e0527586d1d31ee03afc0cb2cd79e02bb52b1e9202a0bc129bc6ac8c89d7L375)
- And for the [latencies](https://github.com/grafana/grafana/pull/50969/files#diff-9c80e0527586d1d31ee03afc0cb2cd79e02bb52b1e9202a0bc129bc6ac8c89d7L415)

**Review by connorlindsey — APPROVED** [review]

**Pull request merged**